pootle/apps/pootle_project/models.py

```diff
@@ -396,6 +395,10 @@ def clean(self):
                 _('"%s" cannot be used as a project code', self.code)
             )
 
+    def directory_exists(self):
```

Comment:
Just a thought here: wouldn't it make sense to implement this for `Directory` models so basically every path object (`Project`, `Language`, `TranslationProject`, `Directory`) gets this method for free?
```python
if self.directory.exists_on_disk():
    do_something()
```

Also note `directory_exists` is ambiguous: it's not clear if it's the in-DB `Directory` or the on-disk directory.